.ipynb_checkpoints/A_Data_collection-checkpoint.ipynb

```diff
@@ -0,0 +1,550 @@
+    "def download_steam_reviews(game_id,default_cursor):\n",
+    "    \n",
+    "    appid=game_id\n",
+    "    base_url = \"https://store.steampowered.com/appreviews/\"\n",
+    "\n",
+    "    # Parameters for the API request\n",
+    "    params = {\n",
+    "        \"json\": 1, \n",
+    "        \"filter\": \"recent\", \n",
+    "        \"language\": \"english\", \n",
+    "        \"review_type\": \"all\", \n",
+    "        \"purchase_type\": \"all\",\n",
+    "        \"num_per_page\": 100,\n",
+    "    }\n",
+    "\n",
+    "    # Store the reviews\n",
+    "    reviews = dict()\n",
+    "\n",
+    "    cursor = default_cursor\n",
+    "\n",
+    "    # Keep track of how many requests are made\n",
+    "    counter = 0\n",
+    "\n",
+    "    # Keep track of how many reviews are fetched\n",
+    "    reviews_fetched = 0\n",
+    "    \n",
+    "    print(\"Starting to send requests...\")\n",
+    "\n",
+    "    while True:\n",
+    "        counter += 1\n",
+    "\n",
+    "        # Append the cursor to the parameters\n",
+    "        params[\"cursor\"] = cursor\n",
+    "\n",
+    "        try:\n",
+    "            response = requests.get(base_url + str(appid), params=params)\n",
+    "\n",
+    "            # Check if the request was successful\n",
+    "            if response.status_code == 200:\n",
+    "\n",
+    "                data = response.json()\n",
+    "\n",
+    "                # Check if the query was successful\n",
+    "                if data[\"success\"] == 1:\n",
+    "                    # Get user reviews\n",
+    "                    reviews_list = data[\"reviews\"]\n",
+    "\n",
+    "                    for review in reviews_list:\n",
+    "\n",
+    "                        recommendationid = review[\"recommendationid\"]\n",
+    "                        review_data = review\n",
+    "\n",
+    "                        reviews[recommendationid] = review_data\n",
+    "\n",
+    "                        reviews_fetched += 1\n",
+    "\n",
+    "                    # Get the cursor value for the next batch of reviews\n",
+    "                    cursor = data[\"cursor\"]\n",
+    "                    # Check if the cursor is empty or None or the same as the last cursor, meaning there are no more reviews\n",
+    "                    if cursor == \"\" or cursor == None or params[\"cursor\"]==cursor:\n",
+    "                        break\n",
+    "\n",
+    "                else:\n",
+    "                    print(\"Query failed\")\n",
+    "                    break\n",
+    "\n",
+    "            else:\n",
+    "                print(\"Request failed\")\n",
+    "                break\n",
+    "\n",
+    "            # Limit the max request to 150 before adding a sleep\n",
+    "            if counter > 9999:\n",
+    "                print(\"Limit of 10000 requests reached. Saving the collected data and the cursor.\")\n",
+    "                with open('cursor_to_start.txt', 'w') as f:\n",
```

Comment:
Also add this in the catch of a try block so that the cursor is saved in the event of a runtime failure, this way you can pick up where you left off without having to sanitize data and redo the batch.